### PR TITLE
Cleanup list fields

### PR DESCRIPTION
**Pull request by mstemm**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area driver-kmod

> /area driver-ebpf

> /area libscap

/area libsinsp

> /area tests

> /area proposals

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Add built-in support for pretty-printing fields that can be used across programs.
```

### CODE REVIEW

**Review by leogr — CHANGES_REQUESTED** [review]
I believe we have to keep the markdown formatting, otherwise SGTM

**Comment by mstemm** [comment]
Sorry I didn't know markdown output was actually used. I added it back. PTAL again!

File: userspace/libsinsp/gen_filter.h [hunk 1]
```diff
@@ -246,21 +246,48 @@ class gen_event_filter_factory
...
 
 		std::list<filter_field_info> fields;
+
+		// Print a terminal-friendly representation of this
+		// field class, including name, description, supported
+		// event sources, and the name and description of each field.
+		std::string as_string(bool verbose, std::set<std::string> event_sources);
+
+		// Calls above with an empty set of event sources.
+		std::string as_string(bool verbose);
```
Comment (FedeDP) [inline userspace/libsinsp/gen_filter.h:281]: Why just not using a default value for above 
`std::string as_string(bool verbose, const std::set<std::string>& event_sources = std::set<std::string>());`

Comment (FedeDP) [reply userspace/libsinsp/gen_filter.h:281]: Moreover, i'd embed the markdown representation there, ie:
`std::string as_string(bool verbose, bool markdown = false, const std::set<std::string>& event_sources = std::set<std::string>());`

File: userspace/libsinsp/fields_info.h [hunk 2]
```diff
@@ -24,5 +24,5 @@ class sinsp;
 //
 // Printer functions
 //
-void list_fields(bool verbose, bool markdown, bool names_only=false);
+void list_fields(bool verbose, bool markdown=false);
```
Comment (FedeDP) [inline userspace/libsinsp/fields_info.h:27]: Is there a reason why names_only support was dropped?
Maybe i am just missing something :)

**Comment by LucaGuerra** [comment]
Tested this change and terminal+markdown formatting seems to be right. Just a little thing: I noticed that `evt (All event types)` appears at the bottom of the list, would it be possible to make it appear on top? This would make the documentation easier to consume when piped to a pager or exported to the documentation website.

**Review by FedeDP — COMMENTED** [review]
I have just a couple of questions, otherwise this looks fine!

File: userspace/libsinsp/fields_info.h (hunk 2 above)
Comment (mstemm) [reply userspace/libsinsp/fields_info.h:27]: This function is only used by oss sysdig, and oss sysdig doesn't actually have an option to only print field names. All the calls to list_fields in the sysdig repo omit the third argument.

Falco has its own function to print field info, and does support a names only argument.

**Comment by mstemm** [comment]
Sorry I didn't know markdown output was actually used. I added it back. PTAL again!

> Tested this change and terminal+markdown formatting seems to be right. Just a little thing: I noticed that `evt (All event types)` appears at the bottom of the list, would it be possible to make it appear on top? This would make the documentation easier to consume when piped to a pager or exported to the documentation website.

There isn't any explicit sorting of fields at the moment, the order is the order in which the filterchecks are added to the global sinsp_filter_check_list: https://github.com/falcosecurity/libs/blob/master/userspace/libsinsp/filter_check_list.cpp#L106. But I did move sinsp_filter_check_gen_event to the top so it will be next to evt (Syscall events only).

**Review by LucaGuerra — APPROVED** [review]
@mstemm yes, I did the same when I wanted to reorder the doc page. LGTM.

**Review by leogr — APPROVED** [review]
/approve

File: userspace/libsinsp/gen_filter.h (hunk 1 above)
Comment (FedeDP) [reply userspace/libsinsp/gen_filter.h]: Thanks! +1 from me

**Review by FedeDP — APPROVED** [review]

**Review by ldegio — APPROVED** [review]

**Pull request merged**